mirgecom/artificial_viscosity.py

```diff
@@ -0,0 +1,328 @@
+
+    # R=Grad(Q) Q flux over interior faces
+    q_flux_int = _facial_flux_q(discr, q_tpair=interior_trace_pair(discr, q))
+    # R=Grad(Q) Q flux interior faces on partition boundaries
+    q_flux_pb = sum(_facial_flux_q(discr, q_tpair=pb_tpair)
+                    for pb_tpair in cross_rank_trace_pairs(discr, q))
+    # R=Grad(Q) Q flux domain boundary part (i.e. BCs)
+    q_flux_db = 0
+    for btag in boundaries:
+        q_tpair = TracePair(
+            btag,
+            interior=discr.project("vol", btag, q),
+            exterior=boundaries[btag].exterior_q(discr, btag=btag, t=t,
+                                                 q=q, eos=eos))
+        q_flux_db = q_flux_db + _facial_flux_q(discr, q_tpair=q_tpair)
+    # Total Q flux across element boundaries
+    q_bnd_flux = q_flux_int + q_flux_pb + q_flux_db
+
+    # Compute R
+    r = discr.inverse_mass(
+        -alpha * indicator * (grad_q_vol - discr.face_mass(q_bnd_flux))
+    )
+
+    # RHS_av = div(R) volume part
+    div_r_vol = discr.weak_div(r)
+    # RHS_av = div(R): grad(Q) flux interior faces part
+    r_flux_int = _facial_flux_r(discr, r_tpair=interior_trace_pair(discr, r))
+    # RHS_av = div(R): grad(Q) flux interior faces on the partition boundaries
+    r_flux_pb = sum(_facial_flux_r(discr, r_tpair=pb_tpair)
+                    for pb_tpair in cross_rank_trace_pairs(discr, r))
+    # RHS_av = div(R): grad(Q) flux domain boundary part (BCs)
+    r_flux_db = 0
+    for btag in boundaries:
+        r_tpair = TracePair(
+            btag,
+            interior=discr.project("vol", btag, r),
+            exterior=boundaries[btag].exterior_grad_q(discr, btag=btag, t=t,
+                                                      grad_q=r, eos=eos))
+        r_flux_db = r_flux_db + _facial_flux_r(discr, r_tpair=r_tpair)
+    # Total grad(Q) flux element boundaries
+    r_flux_bnd = r_flux_int + r_flux_pb + r_flux_db
+
+    # Return the AV RHS term
+    return discr.inverse_mass(-div_r_vol + discr.face_mass(r_flux_bnd))
+
+
+def artificial_viscosity(discr, t, eos, boundaries, q, alpha, **kwargs):
+    """Interface :function:`av_operator` with backwards-compatible API."""
+    from warnings import warn
+    warn("Do not call artificial_viscosity; it is now called av_operator. This"
+         "function will disappear in 2021", DeprecationWarning, stacklevel=2)
+    return av_operator(discr=discr, eos=eos, boundaries=boundaries,
+                       q=q, alpha=alpha, t=t)
+
+
+def compute_smoothness_indicator():
+    """Compute the smoothness indicator for all elements."""
+    from meshmode.array_context import make_loopy_program
+
+    knl = make_loopy_program(
+        """{[iel,idof,j,k]:
+        0<=iel<nelements and
+        0<=idof<ndiscr_nodes_out and
+        0<=j<ndiscr_nodes_in and
+        0<=k<ndiscr_nodes_in}""",
+        "result[iel,idof] = "
+        "sum(k,vec[iel,k]*vec[iel,k]*modes[k])/sum(j,"
+        " vec[iel,j]*vec[iel,j]+1.0e-12/ndiscr_nodes_in)",
```

Comment:
```suggestion
        """
        result[iel,idof] = (
          sum(k,vec[iel,k]*vec[iel,k]*modes[k])
          / sum(j, vec[iel,j]*vec[iel,j]+1.0e-12/ndiscr_nodes_in))
        """,
```
No functional changes (I hope), just better formatting.